File: qa-e2e-automation-suite/src/test/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

import drivers.DriverManager;

public class LoginPage extends BasePage
{
	private WebDriver driver;
	
	private By username = By.id("user-name");
	private By password = By.id("password");
	private By loginButton = By.id("login-button");
	
	public LoginPage()
	{
		this.driver = DriverManager.getDriver();
	}
	
	public void openSite(String url)
	{
		driver.get(url);		
	}
	
	public void login(String user, String pass)
	{
		driver.findElement(username).sendKeys(user);
		driver.findElement(password).sendKeys(pass);
		driver.findElement(loginButton).click();
	}
}

File: EnterpriseAutomationFramework/src/Infrastructure/Web/Pages/LoginPage.cs
using EnterpriseAutomationFramework.Core.Abstractions.Browser;
using EnterpriseAutomationFramework.Core.Enums;
using EnterpriseAutomationFramework.Core.Models;

namespace EnterpriseAutomationFramework.Infrastructure.Web.Pages;

public sealed class LoginPage : BasePage
{
    private readonly ElementLocator _usernameInput = new(LocatorType.Id, "username");

    private readonly ElementLocator _passwordInput = new(LocatorType.Id, "password");

    private readonly ElementLocator _loginButton = 
        new(LocatorType.CssSelector, "button[type='submit']");

    public LoginPage(IBrowser browser) : base(browser)
    { }

    public void GoTo(string url)
    {
        ArgumentException.ThrowIfNullOrWhiteSpace(nameof(url));

        Browser.Navigator.GoTo(url);    
    }

    public void EnterUsername(string username)
    {
        Browser.Actions.Type(_usernameInput, username);
    }

    public void EnterPassword(string password)
    {
        Browser.Actions.Type(_passwordInput, password);
    }

    public void ClickLogin()
    {
        Browser.Actions.Click(_loginButton);
    }

    public void Login(string username, string password)
    {
        EnterUsername(username);
        EnterPassword(password);
        ClickLogin();
    }
}
